sigopt/xgboost/experiment.py

```diff
@@ -36,17 +47,52 @@ def parse_and_create_experiment(self):
         else:
           metric['name'] = DEFAULT_EVALS_NAME + '-' + metric['name']
 
+  def parse_and_create_parameters(self):
+    # Set defaults as needed
+    if 'budget' not in self.experiment_config_parsed:
+      self.experiment_config_parsed['budget'] = DEFAULT_BO_ITERATIONS
```

Comment:
this is not part of `parameter`. I would suggest move this into a separate function.